### ISSUE TITLE
Mission element IDs are incorrectly assigned

### ISSUE THREAD

**Issue by Quackertree** [simon-wh/PAYDAY-2-BeardLib#103]
Hey Luffy!

I believe something minor went wrong in ID assignment when you patched something else. The mission element IDs now also start at 100000; The exact same value that units start at.

This leads to conflicts where elements try to reference units and vice versa, instead of making use of the element/unit they're supposed to refer to.

A patch would be much appreciated. - Manually correcting all these mission elements isn't in my best interest. It could also lead to a lot of confusion for newer custom heisters who cannot get their element sequence to work, even though it seems perfectly fine.

Gratz.

~Gavin

**Comment by Luffyyy** [comment]
Nope you can see that normal maps have that, the game know when it needs to get a unit and when it needs it get an element

**Comment by Quackertree** [comment]
Can you show code that backs this up (or point me to where that code is located)?

I have experienced issues with duplicated IDs in the past and I've found that it was resolved when setting the values of the props to not equal those of the mission elements would actually be the solution to the issues I was having.

This was a long time ago though, so this may also be a combination of remembering it incorrectly and/or having solved the solution by accident, as I was very inexperienced at the time. - I'd just like to see confirmation personally, if possible.

**Comment by Luffyyy** [comment]
Just open any vanilla heist.?

there's a lot of code but mostly if the game needs an element it goes to managers.mission and for units managers.worlddefinition so indexes would never be a problem because they're located in different tables.

**Comment by Quackertree** [comment]
_Tl;dr below; Sry for the long post - Feel free to jump to the conclusion!_

Alrighty, I looked at the code and as far as I can see I cannot find anything that confirms my claim. There are a lot of tables which contain elements, though and they're spread around the place.

However, when I look at the vanilla heists, even though they may _start_ at ID 100000, they are all still _unique_. I've checked over 50 mission script IDs in Nightclub and compared them against the continent file. Both start with ="10...", but every ID is still unique. - I also quickly checked some newer heists, but they all seem to have an actual division where elements start at 200000, for example.

My theory still remains that continent IDs and mission IDs have to be unique, for the following reasons:
1) They are unique in all vanilla heists I have checked, which is equal to 5
2) The newer heists actually split based on continents.continent as the starting ID value. This change wouldn't be neccessary if the "old system" worked fine, most likely indicating that previously, there may have been ID conflicts.
3) I seem to have a strong memory of this being the fix. Not the best argument, but it's there. - I also remember fixing this multiple times across multiple maps.
4) The code itself is spread out a lot, with a bunch of tables of elements spread across a bunch of elements. It would be very easy to miss a scenario where there was an accidental mistake and some tables crossed each other, for example.
5) Even if you are correct and it actually doesn't matter, it's still safer to split them. By not splitting them, you'll never know if there's an issue with duplicate IDs. If you would split the values, you can always guarantee with 100% safety that there will be no duplicate issue between continent & mission IDs.

**Conclusion: Feel free to still ignore the request and close the thread. However, I still recommend you to adjust the base value for elements to be 200000, instead of 100000 as it currently is for above-mentioned reasons. To me, this seems like an easy fix (just changing a number).**

**Comment by Luffyyy** [comment]
Well no, I checked a Raid heist but well still it makes no sense for them to clash, because only elements can call elements and units and the system I built knows each value and what that value needs either units or elements so the ids cannot clash together it's impossible.
